docs/intro.md

```diff
@@ -3,65 +3,176 @@
 that generate custom hardware accelerators.
 
-The following instructions will install all dependencies required to build
-the compiler and run the compiler tests.
+FuTIL has many associated tools and depending on what you are doing, you
+probably only need to install a subset of them.
 
-Start the process by cloning the repository:
+We've organized this document into a core installation that is needed for everything
+and then add-on installations for our command-line driver tool, simulation, frontends,
+and synthesis.
+
+## Core Installation
+
+Start by cloning the repository:
 ```
 git clone https://github.com/cucapra/futil.git
 ```
 
 ### Compiler dependencies
 Install [Rust][rust] (it should automatically install `cargo`).
 
-### Testing dependencies
-The compiler uses expectation tests as well as hardware simulation tests.
+### Install tools for testing
+  1. [runt][] hosts our testing infrastructure. Install with:
+  `cargo install runt`
+  2. [jq][]:
+  Ubuntu: `sudo apt install jq`
+  Mac: `brew install jq`
+  Other platforms: [JQ installation][jq-install]
+
+### Verify installation
+Build the compiler with
+```
+cargo build
+```
+Then run the core tests with:
+```
+runt -i core
+```
+
+If everything has been installed correctly, this should not produce any failing
+tests.
+
+## Add-on Installations
+Here are instructions for optional add-ons. We recommend installing at least `fud`.
+### Fud (Command-line driver) [Recommended]
+<details><summary>Click to expand!</summary>
+
+[The FuTIL driver](./tools/fud.md) is a command line tool to drive the FuTIL
+compiler and coordinate invoking frontends and backends.
+
+You need [Flit][flit] to install `fud`.
+```
+pip3 install flit
+```
+
+Then install `fud` with:
+```
+cd fud
+flit install -s
+```
+
+`fud` needs to know where the FuTIL directory lives. Running `fud check` will ask you
+for the current directory and then display information about the tools that it could find.
+```
+fud check
+```
+</details>
+
+### Simulation backend (Verilator)
+<details><summary>Click to expand!</summary>
 
-1. [runt][] by running `cargo install runt`
-2. [vcdump][] by running `cargo install vcdump`
-3. [Verilator][]:
-    - Ubuntu: `sudo apt install verilator`
-    - Fedora: `sudo dnf install verilator`
-    - Mac: `brew install verilator`
-    - Other platforms: [Verilator installation][verilator-install].
-4. [jq][]:
-    - Ubuntu: `sudo apt install jq`
-    - Fedora: `sudo dnf install jq`
-    - Mac: `brew install jq`
-    - Other platforms: [JQ installation][jq-install].
-5. (Optional) [flit][]:
-    - `python -m pip install flit`
+We use [Verilator][verilator] to simulate compiled designs and verify correctness. If you're on a Mac,
+install with:
+```
+brew install verilator
+```
 
-### Install the FuTIL Driver
+Otherwise, you will probably need to compile it from source yourself (the versions in Linux repositories are generally out-of-date.)
```

Comment:
```suggestion
Otherwise, you will probably need to compile it from source yourself (the versions in Linux repositories are generally out of date).
```